### PR TITLE
Setup GitHub Infrastructure

### PR DESCRIPTION
**Pull request by stormc**
Add some GitHub infrastructure and adapt existing documentation to apply to Specifications.

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,19 @@
+### Description
+
+Please provide a concise summary of the changes and their motivation.
+
+#### Issues Addressed
+
+List and link all the issues addressed by this PR.
+
+#### Change Type
```
Comment (jachstet-sea) [inline .github/pull_request_template.md:9]: Not sure if we need that here if we can use Labels instead/in addition?

**Review by jachstet-sea — APPROVED** [review]
lgtm except for one comment I posted inline

File: .github/pull_request_template.md (hunk 1 above)
Comment (stormc) [reply .github/pull_request_template.md:9]: Well, I would use labels in addition if we need that structuring. Labels are manually attached as well and this is a hint to place the right one on the PR. Granted, you can automate PR labeling, e.g., with [labeler](https://github.com/actions/labeler) or [auto-label-pulls](https://github.com/shiftEscape/auto-label-pulls), but those labeling actions need context information to infer what label(s) to attach. This requires some strict rules or namings applied which we don't have in the `specification` repository.

TL;DR: I'm opting for keeping it so the PR author hints to how this PR is to be processed.

File: .github/ISSUE_TEMPLATE/1-enhancement.yml [hunk 2]
```diff
@@ -0,0 +1,32 @@
+---
+name: Enhancement request
+description: Request an enhancement for Margo Specification
```
Comment (phil-abb) [inline .github/ISSUE_TEMPLATE/1-enhancement.yml]: would "to the" work better than "for" here?

File: .github/ISSUE_TEMPLATE/2-bug.yml [hunk 3]
```diff
@@ -0,0 +1,30 @@
+---
+name: Bug report
+description: File a new bug to Margo Specification
```
Comment (phil-abb) [inline .github/ISSUE_TEMPLATE/2-bug.yml]: would "for the" work better than "to" here?

**Review by phil-abb — COMMENTED** [review]
Made acouple of recommendations for updating text.

File: .github/ISSUE_TEMPLATE/1-enhancement.yml (hunk 2 above)
Comment (stormc) [reply .github/ISSUE_TEMPLATE/1-enhancement.yml]: Yes, fixed.

File: .github/ISSUE_TEMPLATE/2-bug.yml (hunk 3 above)
Comment (stormc) [reply .github/ISSUE_TEMPLATE/2-bug.yml]: Yes, fixed.

**Review by phil-abb — APPROVED** [review]
